aspnet/mvc/overview.rst

```diff
@@ -1,8 +1,313 @@
+
+.. code-block:: C#
+
+  public async Task<IActionResult> Login(LoginViewModel model, string returnUrl = null) { ... }
+
+Model validation
+^^^^^^^^^^^^^^^^
+
+In addition to binding request data to model objects, ASP.NET Core MVC provides a powerful :doc:`validation <mvc/models/validation>` system. Simply add validation attributes to your model types using data annotation attributes and then check for model errors in your controller action.
+
+.. code-block:: c#
+  :emphasize-lines: 4-5,8-9
+
+  using System.ComponentModel.DataAnnotations;
+  public class LoginViewModel
+  {
+      [Required]
+      [EmailAddress]
+      public string Email { get; set; }
+
+      [Required]
+      [DataType(DataType.Password)]
+      public string Password { get; set; }
+
+      [Display(Name = "Remember me?")]
+      public bool RememberMe { get; set; }
+  }
+
+A controller action:
+
+.. code-block:: c#
+  :emphasize-lines: 3
+  
+  public async Task<IActionResult> Login(LoginViewModel model, string returnUrl = null)
+  {
+      if (ModelState.IsValid)
+      {
+        // work with the model
+      }
+      // If we got this far, something failed, redisplay form
+      return View(model);
+  }
+
+The framework will handle validating request data both on the client and on the server. Validation logic specified on model types is added to the rendered views as unobtrusive annotations and is enforced in the browser via `jQuery Validation <http://jqueryvalidation.org/>`__.
+
+Dependency injection
+^^^^^^^^^^^^^^^^^^^^
+
+ASP.NET Core MVC leverages ASP.NET Core's built-in support for :doc:`dependency injection </fundamentals/dependency-injection>`. :doc:`Controllers </mvc/controllers/dependency-injection>` can request needed services through their constructors, allowing them to follow the `Explicit Dependencies Principle <http://deviq.com/explicit-dependencies-principle/>`_:
+
+.. literalinclude:: /../common/samples/WebApplication1/src/WebApplication1/Controllers/AccountController.cs
+  :lines: 17-38
+  :emphasize-lines: 10-14
+  :language: c#
+
+Your app can also use :doc:`dependency injection in view files </mvc/views/dependency-injection>`, using the ``@inject`` directive:
+
+.. code-block:: html
+  :emphasize-lines: 1
+
+  @inject SomeService ServiceName
+  <!DOCTYPE html>
+  <html>
+  <head>
+    <title>@ServiceName.GetTitle</title>
+  </head>
+  <body>
+    <h1>@ServiceName.GetTitle</h1>
+  </body>
+  </html>
+
+Filters
+^^^^^^^
+
+:doc:`Filters </mvc/controllers/filters>` help developers encapsulate cross-cutting concerns, like exception handling or authorization. Filters enable running custom pre- and post-processing logic for action methods, and can be configured to run at certain points within the execution pipeline for a given request. Filters can be applied to controllers or actions as attributes, and several filters (such as ``Authorize``) are included in the framework.
```

Comment:
Filters can be applied to controllers or actions as attributes.  Several filters (such as ``Authorize``) are included in the framework.